modules/utxo-coredao/src/opReturn.ts

```diff
@@ -0,0 +1,214 @@
+  // if timelock is provided, write it into 32-bit little-endian
+  const timelockBuffer = 'timelock' in rest ? encodeTimelock(rest.timelock) : Buffer.from([]);
+  const data = Buffer.concat([
+    CORE_DAO_SATOSHI_PLUS_IDENTIFIER,
+    versionBuffer,
+    chainId,
+    delegator,
+    validator,
+    feeBuffer,
+    redeemScriptBuffer,
+    timelockBuffer,
+  ]);
+  if (data.length > 80) {
+    throw new Error('OP_RETURN outputs cannot have a length larger than 80 bytes');
+  }
+
+  const payment = payments.embed({
+    data: [data],
+    network: chainId.equals(CORE_DAO_TESTNET_CHAIN_ID) ? networks.testnet : networks.bitcoin,
+  });
+  if (!payment.output) {
+    throw new Error('Unable to create OP_RETURN output');
+  }
+
+  return payment.output;
+}
+
+/**
+ * Parse a CoreDAO OP_RETURN output script into the constituent parts
+ * @param script
+ * @returns OpReturnParams
+ */
+export function parseCoreDaoOpReturnOutputScript(script: Buffer): OpReturnParams {
+  if (!script.subarray(0, 1).equals(OP_RETURN_IDENTIFIER)) {
+    throw new Error('First byte must be an OP_RETURN');
+  }
+
+  const payment = payments.embed({
+    output: script,
+  });
+  const data = payment.data;
+  if (!data || data.length !== 1) {
+    throw new Error('Invalid OP_RETURN output');
+  }
+  const dataBuffer = data[0];
+  if (dataBuffer.length > 80) {
+    throw new Error(`OP_RETURN outputs cannot have a length larger than 80 bytes`);
+  }
+  let offset = 0;
+
+  // Decode satoshi+ identifier
+  if (!dataBuffer.subarray(offset, offset + 4).equals(CORE_DAO_SATOSHI_PLUS_IDENTIFIER)) {
+    throw new Error('Invalid satoshi+ identifier');
+  }
+  offset += 4;
+
+  // Decode version
+  const version = dataBuffer[offset];
+  offset += 1;
+
+  // Decode chainId
+  const chainId = Buffer.from(dataBuffer.subarray(offset, offset + 2));
+  if (!(chainId.equals(CORE_DAO_TESTNET_CHAIN_ID) || chainId.equals(CORE_DAO_MAINNET_CHAIN_ID))) {
+    throw new Error(
+      `Invalid ChainID: ${chainId.toString('hex')}. Must be either 0x045b (testnet) or 0x045c (mainnet).`
+    );
+  }
+  offset += 2;
+
+  // Decode delegator
+  const delegator = Buffer.from(dataBuffer.subarray(offset, offset + 20));
+  offset += 20;
+
+  // Decode validator
+  const validator = Buffer.from(dataBuffer.subarray(offset, offset + 20));
+  offset += 20;
+
+  // Decode fee
+  const fee = dataBuffer[offset];
+  offset += 1;
+
+  const baseParams = { version, chainId, delegator, validator, fee };
+
+  // Decode redeemScript or timelock
+  if (offset === dataBuffer.length - 4) {
+    return { ...baseParams, timelock: decodeTimelock(dataBuffer.subarray(offset)) };
+  } else {
+    return { ...baseParams, redeemScript: Buffer.from(dataBuffer.subarray(offset)) };
+  }
+}
+
+/**
+ * Check that the params are valid and that they make the given script
+ * @param script
+ * @param params
+ */
+export function verifyCoreDaoOpReturnOutputScript(script: Buffer, params: OpReturnParams): boolean {
```

Comment:
this hides the exception details which can be crucial

it looks like this is just a testutil at the moment so just move it there

if you think it is worth in the main code please change it to

```suggestion
export assertStandardCoreDaoOpReturnOutputScript(script: Buffer, params: OpReturnParams): void {
```